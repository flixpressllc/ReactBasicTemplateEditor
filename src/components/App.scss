@import "../styles/base-includes";

.reactBasicTemplateEditor-App {
  @extend %base-container-styles;
  padding-top: iH($xs-image);
  background-color: #ddd;
  height:100%;

  @include breakpoint($sm) {
    padding-top: iH($sm-image);
  }

  @include breakpoint($md) {
    padding-top: ($md-title-font-size * $line-height) + ($medPad * 2);
  }
}

.reactBasicTemplateEditor-App-title {
  position: fixed;
  color: #fff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  top: 0;
  left: 0;
  right: 0;
  padding: $medPad;
  padding-right: iW($xs-image) + 8;
  min-height: iH($xs-image);
  background-color: $flixpress-blue;
  z-index: 1;

  @include breakpoint($sm) {
    font-size: $sm-title-font-size;
    font-weight: bold;
    padding-right: iW($sm-image) + 8;
    min-height: iH($sm-image);
  }

  @include breakpoint($md) {
    min-height: 0;
    padding-right: $medPad;
  }
}

@include breakpoint($md) {
  .reactBasicTemplateEditor-App-formArea {
    position: relative;
    max-width: 850px;
    margin-left: auto;
    margin-right: auto;
    @include pie-clearfix;
  }

  .reactBasicTemplateEditor-App-column {
    width: 49%;
    float: left;
    margin: 15px 0.5% 0;

    &:first-child {
      padding-right: $viewportPad / 2;
    }

    &:last-child {
      padding-left: $viewportPad / 2;
      // Make room for image
      margin-top: $medPad + iH($md-image);
    }
  }
}


  .reactBasicTemplateEditor-App-column-right {
    width: 49%;
    float: right;
    margin: 15px 0.5% 0;
    position: fixed;

    &:first-child {
      padding-right: $viewportPad / 2;
    }

    &:last-child {
      padding-left: $viewportPad / 2;
      // Make room for image
      margin-top: $bigPad * 2 + iH($md-image);
    }
  }

.reactBasicTemplateEditor-App-submissionModalSuccess {
  @extend %confirm-modal;
}

.reactBasicTemplateEditor-App-submissionModalSuccessOverlay {
  @extend %confirm-overlay;
}
