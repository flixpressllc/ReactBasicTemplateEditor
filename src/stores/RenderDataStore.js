import { EventEmitter } from 'events';
import dispatcher from './dispatcher';
import * as Containers from '../utils/globalContainerConcerns';

import { clone, isObject } from '../utils/helper-functions';

class RenderDataStore extends EventEmitter {
  constructor () {
    super();
    this.state = {
      receivedInitial: false
    }
  }
  changeContainer(dataType, fieldName, newData) {
    let containerName = Containers.getContainerNameFor(dataType);
    Object.assign(this.containers[containerName][fieldName], newData);
    this.emit('change', {containerName, dataType, fieldName});
  }
  
  addMissingContainers() {
    if (!isObject(this.containers)) {
      this.containers = {};
    }
    Containers.getContainerNames().map(name => {
      if (!this.containers.hasOwnProperty(name)) {
        this.containers[name] = {};
      }
    })
  }

  getAll() {
    if ( ! this.state.receivedInitial) this.addMissingContainers();
    return this.containers ? clone(this.containers) : {};
  }
  
  handleActions(action) {
    switch(action.type) {
      case 'CHANGE_CONTAINER':
        this.changeContainer(action.dataType, action.fieldName, action.newData);
      break;
      case 'RECEIVE_INITIAL_CONTAINER_VALUES':
        this.state.receivedInitial = true;
        this.containers = action.containers;
        this.emit('change');
      break;
    }
  }
}

const renderDataStore = new RenderDataStore;
dispatcher.register(renderDataStore.handleActions.bind(renderDataStore))
window.renderDataStore = renderDataStore;
export default renderDataStore;
