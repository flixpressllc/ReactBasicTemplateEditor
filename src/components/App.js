import React from 'react';
import * as TemplateSpecActions from '../actions/TemplateSpecActions';
import * as TemplateOptionsActions from '../actions/TemplateOptionsActions';
import * as StateActions from '../actions/StateActions';
import { find } from '../utils/dom-queries';
import { adjustColorbox } from '../utils/colorbox-manipulation';

import Messages from './UserMessages';
import SubmitRender from './SubmitRender';
import ResolutionPicker from './ResolutionPicker';
import SpecFields from './SpecFields';
import SoundPicker from './SoundPicker';
import Modal from './lib/Modal';

// This must be called after all the actual containers are called so they can
// register themselves before RenderDataStore inside dataLayer tries to get them all...
import RenderDataStore from '../stores/RenderDataStore';
import dl from '../utils/dataLayer';

import StateStore from '../stores/StateStore';

import './App.scss';

class App extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      allowSubmit: false,
      caughtErrors: []
    };

    StateActions.setState({
      templateType: props.templateType,
      templateId: parseInt(props.templateId, 10),
      username: props.userSettingsData.username,
      imageAspectRatio: props.imageAspectRatio
    });

    this.handlePlaceOrder = this.handlePlaceOrder.bind(this);
    this.handlePlacePreviewOrder = this.handlePlacePreviewOrder.bind(this);
    this.handleChooseSong = this.handleChooseSong.bind(this);
    this.handlePreviewChange = this.handlePreviewChange.bind(this);
    this.handleResolutionIdChange = this.handleResolutionIdChange.bind(this);
    this.handleUpdateCaughtErrors = this.handleUpdateCaughtErrors.bind(this);
    this.handleUpdateTemplateOptions = this.handleUpdateTemplateOptions.bind(this);
    this.editorSetupDidComplete = this.editorSetupDidComplete.bind(this);

    StateStore.on('STATE_UPDATED', this.handleUpdateCaughtErrors);

    RenderDataStore.on('TEMPLATE_OPTIONS_CHANGED', this.handleUpdateTemplateOptions);
  }

  handleUpdateTemplateOptions () {
    this.forceUpdate();
  }

  handlePlacePreviewOrder () {
    TemplateOptionsActions.setTemplateOptions({isPreview: true});
    this.handlePlaceOrder();
  }

  handleUpdateCaughtErrors () {
    this.setState(StateStore.getState(['caughtErrors']));
  }

  setupEditor () {
    return dl.setupEditor(this.props.uiSettingsJsonUrl);
  }

  componentDidMount () {
    TemplateSpecActions.setSpecs({templateId: parseInt(this.props.templateId, 10)});
    this.setupEditor().then(this.editorSetupDidComplete);
  }

  editorSetupDidComplete () {
    // This doesn't mean that the images and such have acutally loaded,
    // so the height of the editor is not yet determined, etc.
    this.setState( {editorSetupIsComplete: true} );
    setTimeout(() => adjustColorbox(), 500)
  }

  componentWillUnmount () {
    StateStore.removeEventListener('STATE_UPDATED', this.handleUpdateCaughtErrors);
    RenderDataStore.removeEventListener('TEMPLATE_OPTIONS_CHANGED', this.handleUpdateTemplateOptions);
  }

  handleResolutionIdChange (id) {
    TemplateOptionsActions.setTemplateOptions({resolutionId: id});
  }

  handlePreviewChange (e) {
    TemplateOptionsActions.setTemplateOptions({isPreview: e.target.checked});
  }

  handlePlaceOrder () {

    if (dl.prepOrderForSubmit()) {
      this.setState({allowSubmit: true}, function () {
        setTimeout(function(){ find('form input[type="submit"]')[0].click(); }, 100);
      });
    } else {
      throw new Error('Submission failed.');
    }
  }

  handleChooseSong (audioInfo) {
    TemplateOptionsActions.setTemplateOptions({audioInfo});
  }

  render() {
    var resolutionPicker = null;
    if (this.state.editorSetupIsComplete) {
      resolutionPicker = (
        <ResolutionPicker
          resolutionOptions={RenderDataStore.getTemplateOptions('resolutionOptions')}
          resolutionIdChange={this.handleResolutionIdChange}
          chosen={RenderDataStore.getTemplateOptions('resolutionId')}
          disabled={RenderDataStore.getTemplateOptions('isPreview')}
        />
      );
    }
    var specFields = null;
    if (this.state.editorSetupIsComplete) {
      specFields = (
        <SpecFields
          templateType={ StateStore.getState('templateType')}
          uiSections={RenderDataStore.getUiDefinition()}
          imageBank={RenderDataStore.getTemplateOptions('imageBank')}
        />
      );
    }

    return (
      <div className="reactBasicTemplateEditor-App">
        <h1 className="reactBasicTemplateEditor-App-title">
          <span>Template {this.props.templateId}</span>
        </h1>
        <Messages messages={this.state.caughtErrors} typeOverride="bad"/>
        <div className="reactBasicTemplateEditor-App-formArea">
            <div className="reactBasicTemplateEditor-App-column">
            { specFields }
          </div>
          <div className="reactBasicTemplateEditor-App-column">
            <SoundPicker
              audioInfo={RenderDataStore.getTemplateOptions('audioInfo')}
              username={this.props.userSettingsData.username}
              onChooseSong={this.handleChooseSong}
            />
            {resolutionPicker}
            <SubmitRender
              isPreview={RenderDataStore.getTemplateOptions('isPreview')}
              onChange={this.handlePreviewChange}
              placeOrder={this.handlePlaceOrder}
              allowSubmit={this.state.allowSubmit}
              userSettingsData={this.props.userSettingsData}
              placePreviewOrder={this.handlePlacePreviewOrder}
            />
          </div>
        </div>
        <Modal
          isOpen={this.state.allowSubmit}
          className="reactBasicTemplateEditor-App-submissionModalSuccess"
          overlayClassName="reactBasicTemplateEditor-App-submissionModalSuccessOverlay"
          contentLabel="Submission Modal"
        >
          Your order is being submitted.
        </Modal>
      </div>
    );
  }
}

export default App;
