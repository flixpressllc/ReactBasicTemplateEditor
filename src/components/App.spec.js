import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import RenderDataStore from '../stores/RenderDataStore';

jest.mock('../utils/renderDataAdapter');

jest.mock('../utils/ajax');
const ajax = require('../utils/ajax');
ajax.getJSON = (path) => {
  return new Promise ((resolve, reject) => {
    require('jsonfile').readFile('./src' + path, (err, obj) => {
      if (err) reject(err);
      resolve({data: obj});
    });
  });
};

function generalSettings () { return {
  uiSettingsJsonUrl: '/templates/Template1000.json',
  templateType: 'text',
  templateId: 1000,
  userSettingsData: {
    username: 'DonDenton',
    minutesRemainingInContract: 170.2819,
    minimumTemplateDuration: 0.1667,
    isChargePerOrder: false,
    renderCost: 8
  }
};}

function generalImagesTemplateSettings () { return {
  uiSettingsJsonUrl: '/templates/Template2000.json',
  templateType: 'images',
  templateId: 2000,
  userSettingsData: {
    username: 'DonDenton',
    minutesRemainingInContract: 170.2819,
    minimumTemplateDuration: 0.1667,
    isChargePerOrder: false,
    renderCost: 8
  }
};}

function baseMockReturnForGetStartingData () { return [{
  isPreview: false,
  resolutionId: 0,
  resolutions: [
    {
      id: 5,
      name: '720p'
    },
    {
      id: 3,
      name: '1080p'
    },
    {
      id: 4,
      name: '4K'
    }
  ],
  audioInfo: {
    audioType: 'NoAudio',
    audioUrl: '',
    id: 0,
    length: 0,
    name: ''
  }
}];}

function imagesMockReturnForGetStartingData () {
  return [Object.assign({}, baseMockReturnForGetStartingData()[0], {
    imageBank: [
      'DonDentonAdmin_1-23-2017_94956756.png',
      'DonDentonAdmin_1-23-2017_9502787.jpg',
      'DonDentonAdmin_1-23-2017_9505506.png'
    ]
  }),{}];
}

function baseMockReturnForGetStartingDataOffPreview () { return [
  {
    'audioInfo': {
      'audioType': 'StockAudio',
      'audioUrl': 'https://fpsound.s3.amazonaws.com/13.mp3',
      'id': 13,
      'length': 3,
      'name': 'Bunny Garden'
    },
    'isPreview': true,
    'resolutionId': 5,
    'resolutions': [
      {
        'id': 5,
        'name': '720p'
      },
      {
        'id': 3,
        'name': '1080p'
      }
    ]
  },
  {
    'Icon Style': 'camera',
    'Icon Character - type in only one (optional)': 'q',
    'YT1': 'The Office Audition Tapes For Dwight, Michael, Kevin, Pam and Jim|S73Nzksy6rU|',
    'Icon Color': 'icon_pink',
    'Main Text Color': '100.100.0',
    'Subtitle Text Color': '0.0.0.50',
    'Text Left of Icon': 'Left',
    'Text Right of Icon': 'Right',
    'Subtitle': 'Bottom'
  }
];}

function imagesMockReturnForGetStartingDataOffPreview () { return [
  {
    audioInfo: {
      audioType: 'StockAudio',
      audioUrl: 'https://fpsound.s3.amazonaws.com/13.mp3',
      id: 13,
      length: 3,
      name: 'Bunny Garden'
    },
    isPreview: true,
    resolutionId: 5,
    resolutions: [
      {
        id: 5,
        name: '720p'
      },
      {
        id: 3,
        name: '1080p'
      }
    ]
  },
  {
    'Text Left of Icon': 'Left',
    'Text Right of Icon': 'Right',
    ImageContainer: [ // This name will be wrong if we ever assume more than one
      {
        captions: [ 'marmet' ],
        file: 'DonDentonAdmin_1-23-2017_94956756.png'
      },
      {
        captions: [''],
        file: 'DonDentonAdmin_1-23-2017_9502787.jpg'
      },
      {
        captions: [''],
        file: 'DonDentonAdmin_1-23-2017_9505506.png'
      }
    ]
  }
];}

describe('App', () => {
  it('renders without crashing', () => {
    const settings = generalSettings();
    expect(() => {
      shallow(<App {...settings}/>);
    }).not.toThrow();
  });

  describe('when text only', () => {
    describe('and when starting fresh', () => {
      it('prepares the data for order as expected', () => {
        let settings = generalSettings();
        let renderDataAdapter = require('../utils/renderDataAdapter');
        renderDataAdapter.updateXmlForOrder.mockImplementation(() => {});
        renderDataAdapter.getReactStartingData.mockImplementation(() => baseMockReturnForGetStartingData() );
        const component = shallow(<App {...settings}/>);

        return component.instance().setupEditor().then(() => {
          component.instance().handlePlaceOrder();

          let numCalls = renderDataAdapter.updateXmlForOrder.mock.calls.length;
          expect(renderDataAdapter.updateXmlForOrder.mock.calls[numCalls - 1]).toMatchSnapshot();
        });
      });
    });
    describe('and when starting with a preview', () => {
      it('prepares the data for order as expected', () => {
        let settings = generalSettings();
        let renderDataAdapter = require('../utils/renderDataAdapter');
        renderDataAdapter.updateXmlForOrder.mockImplementation(() => {});
        renderDataAdapter.getReactStartingData.mockImplementation(() => baseMockReturnForGetStartingDataOffPreview() );
        const component = shallow(<App {...settings}/>);

        return component.instance().setupEditor().then(() => {
          component.instance().handlePlaceOrder();

          let numCalls = renderDataAdapter.updateXmlForOrder.mock.calls.length;
          expect(renderDataAdapter.updateXmlForOrder.mock.calls[numCalls - 1]).toMatchSnapshot();
        });
      });
    });
  });

  describe('when an images template', () => {
    describe('and when starting fresh', () => {
      it('passes the info into the correct containers', () => {
        let settings = generalImagesTemplateSettings();
        let renderDataAdapter = require('../utils/renderDataAdapter');
        renderDataAdapter.updateXmlForOrder.mockImplementation(() => {});
        renderDataAdapter.getReactStartingData.mockImplementation(() => imagesMockReturnForGetStartingData() );
        const component = shallow(<App {...settings}/>);

        return component.instance().setupEditor().then(() => {
          let containedImages = RenderDataStore.getAll().userImageChoosers['Your Two Images'].containedImages;

          let expected = [
            {id: 0, file: 'DonDentonAdmin_1-23-2017_94956756.png', captions: ['','','']},
            {id: 1, file: 'DonDentonAdmin_1-23-2017_9502787.jpg', captions: ['','','']},
            {id: 2, file: 'DonDentonAdmin_1-23-2017_9505506.png', captions: ['','','']}
          ];

          expect(containedImages).toEqual(expected);
        });
      });
      it('prepares the data for order as expected', () => {
        let settings = generalImagesTemplateSettings();
        let renderDataAdapter = require('../utils/renderDataAdapter');
        renderDataAdapter.updateXmlForOrder.mockImplementation(() => {});
        renderDataAdapter.getReactStartingData.mockImplementation(() => imagesMockReturnForGetStartingData() );
        const component = shallow(<App {...settings}/>);

        return component.instance().setupEditor().then(() => {
          component.instance().handlePlaceOrder();

          let numCalls = renderDataAdapter.updateXmlForOrder.mock.calls.length;
          expect(renderDataAdapter.updateXmlForOrder.mock.calls[numCalls - 1]).toMatchSnapshot();
        });
      });
    });

    describe('and when starting from preview', () => {
      it('passes the info into the correct containers', () => {
        let settings = generalImagesTemplateSettings();
        let renderDataAdapter = require('../utils/renderDataAdapter');
        renderDataAdapter.updateXmlForOrder.mockImplementation(() => {});
        renderDataAdapter.getReactStartingData.mockImplementation(() => imagesMockReturnForGetStartingDataOffPreview() );
        const component = shallow(<App {...settings}/>);

        return component.instance().setupEditor().then(() => {
          let containedImages = RenderDataStore.getAll().userImageChoosers['Your Two Images'].containedImages;
          expect(containedImages).toMatchSnapshot();
        });
      });
      it('prepares the data for order as expected', () => {
        let settings = generalImagesTemplateSettings();
        let renderDataAdapter = require('../utils/renderDataAdapter');
        renderDataAdapter.updateXmlForOrder.mockImplementation(() => {});
        renderDataAdapter.getReactStartingData.mockImplementation(() => imagesMockReturnForGetStartingDataOffPreview() );
        const component = shallow(<App {...settings}/>);

        return component.instance().setupEditor().then(() => {
          component.instance().handlePlaceOrder();

          let numCalls = renderDataAdapter.updateXmlForOrder.mock.calls.length;
          expect(renderDataAdapter.updateXmlForOrder.mock.calls[numCalls - 1]).toMatchSnapshot();
        });
      });
    });
  });
});
