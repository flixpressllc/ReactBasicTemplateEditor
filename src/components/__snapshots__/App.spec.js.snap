// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App when an images template and when starting fresh prepares the data for order as expected 1`] = `
Array [
  Object {
    "audioInfo": Object {
      "audioType": "NoAudio",
      "audioUrl": "",
      "id": 0,
      "length": 0,
      "name": "",
    },
    "imageBank": Array [
      "DonDentonAdmin_1-23-2017_94956756.png",
      "DonDentonAdmin_1-23-2017_9502787.jpg",
      "DonDentonAdmin_1-23-2017_9505506.png",
    ],
    "isPreview": false,
    "resolutionId": 0,
    "resolutionOptions": Array [
      Object {
        "id": 5,
        "name": "720p",
      },
      Object {
        "id": 3,
        "name": "1080p",
      },
      Object {
        "id": 4,
        "name": "4K",
      },
    ],
    "ui": Array [
      Object {
        "Customize Text": Array [
          Object {
            "name": "Text Left of Icon",
            "type": "TextField",
            "value": "",
          },
          Object {
            "name": "Text Right of Icon",
            "type": "TextField",
            "value": "",
          },
        ],
      },
      Object {
        "Customize Images": Array [
          Object {
            "name": "Your Two Images",
            "type": "UserImageChooser",
            "value": Array [
              Object {
                "captions": Array [
                  Object {
                    "label": "Top Text",
                    "value": "",
                  },
                  Object {
                    "label": "Middle Text",
                    "value": "",
                  },
                  Object {
                    "label": "Bottom Text",
                    "value": "",
                  },
                ],
                "file": "DonDentonAdmin_1-23-2017_94956756.png",
                "id": 0,
              },
              Object {
                "captions": Array [
                  Object {
                    "label": "Top Text",
                    "value": "",
                  },
                  Object {
                    "label": "Middle Text",
                    "value": "",
                  },
                  Object {
                    "label": "Bottom Text",
                    "value": "",
                  },
                ],
                "file": "DonDentonAdmin_1-23-2017_9502787.jpg",
                "id": 1,
              },
              Object {
                "captions": Array [
                  Object {
                    "label": "Top Text",
                    "value": "",
                  },
                  Object {
                    "label": "Middle Text",
                    "value": "",
                  },
                  Object {
                    "label": "Bottom Text",
                    "value": "",
                  },
                ],
                "file": "DonDentonAdmin_1-23-2017_9505506.png",
                "id": 2,
              },
            ],
          },
        ],
      },
    ],
  },
]
`;

exports[`App when an images template and when starting from preview passes the info into the correct containers 1`] = `
Array [
  Object {
    "captions": Array [
      "marmet",
    ],
    "file": "DonDentonAdmin_1-23-2017_94956756.png",
    "id": 0,
  },
  Object {
    "captions": Array [
      "",
    ],
    "file": "DonDentonAdmin_1-23-2017_9502787.jpg",
    "id": 1,
  },
  Object {
    "captions": Array [
      "",
    ],
    "file": "DonDentonAdmin_1-23-2017_9505506.png",
    "id": 2,
  },
]
`;

exports[`App when an images template and when starting from preview prepares the data for order as expected 1`] = `
Array [
  Object {
    "audioInfo": Object {
      "audioType": "StockAudio",
      "audioUrl": "https://fpsound.s3.amazonaws.com/13.mp3",
      "id": 13,
      "length": 3,
      "name": "Bunny Garden",
    },
    "imageBank": Array [
      "DonDentonAdmin_1-23-2017_94956756.png",
      "DonDentonAdmin_1-23-2017_9502787.jpg",
      "DonDentonAdmin_1-23-2017_9505506.png",
    ],
    "isPreview": true,
    "resolutionId": 5,
    "resolutionOptions": Array [
      Object {
        "id": 5,
        "name": "720p",
      },
      Object {
        "id": 3,
        "name": "1080p",
      },
    ],
    "ui": Array [
      Object {
        "Customize Text": Array [
          Object {
            "name": "Text Left of Icon",
            "type": "TextField",
            "value": "Left",
          },
          Object {
            "name": "Text Right of Icon",
            "type": "TextField",
            "value": "Right",
          },
        ],
      },
      Object {
        "Customize Images": Array [
          Object {
            "name": "Your Two Images",
            "type": "UserImageChooser",
            "value": Array [
              Object {
                "captions": Array [
                  Object {
                    "label": "Top Text",
                    "value": "marmet",
                  },
                  Object {
                    "label": "Middle Text",
                    "value": "",
                  },
                  Object {
                    "label": "Bottom Text",
                    "value": "",
                  },
                ],
                "file": "DonDentonAdmin_1-23-2017_94956756.png",
                "id": 0,
              },
              Object {
                "captions": Array [
                  Object {
                    "label": "Top Text",
                    "value": "",
                  },
                  Object {
                    "label": "Middle Text",
                    "value": "",
                  },
                  Object {
                    "label": "Bottom Text",
                    "value": "",
                  },
                ],
                "file": "DonDentonAdmin_1-23-2017_9502787.jpg",
                "id": 1,
              },
              Object {
                "captions": Array [
                  Object {
                    "label": "Top Text",
                    "value": "",
                  },
                  Object {
                    "label": "Middle Text",
                    "value": "",
                  },
                  Object {
                    "label": "Bottom Text",
                    "value": "",
                  },
                ],
                "file": "DonDentonAdmin_1-23-2017_9505506.png",
                "id": 2,
              },
            ],
          },
        ],
      },
    ],
  },
]
`;

exports[`App when text only and when starting fresh prepares the data for order as expected 1`] = `
Array [
  Object {
    "audioInfo": Object {
      "audioType": "NoAudio",
      "audioUrl": "",
      "id": 0,
      "length": 0,
      "name": "",
    },
    "imageBank": Array [],
    "isPreview": false,
    "resolutionId": 0,
    "resolutionOptions": Array [
      Object {
        "id": 5,
        "name": "720p",
      },
      Object {
        "id": 3,
        "name": "1080p",
      },
      Object {
        "id": 4,
        "name": "4K",
      },
    ],
    "ui": Array [
      Object {
        "Customize Icon": Array [
          Object {
            "name": "Icon Style",
            "type": "DropDown",
            "value": "MyOwnCharacter",
          },
          Object {
            "name": "Icon Character - type in only one (optional)",
            "type": "TextField",
            "value": "",
          },
        ],
      },
      Object {
        "Customize Look": Array [
          Object {
            "name": "YT1",
            "type": "YouTubeLink",
            "value": "",
          },
          Object {
            "name": "Icon Color",
            "type": "DropDown",
            "value": "icon_black",
          },
          Object {
            "name": "Main Text Color",
            "type": "ColorPicker",
            "value": "0.0.0",
          },
          Object {
            "name": "Subtitle Text Color",
            "type": "ColorPicker",
            "value": "256.100.100.50",
          },
        ],
      },
      Object {
        "Customize Text": Array [
          Object {
            "name": "Text Left of Icon",
            "type": "TextField",
            "value": "",
          },
          Object {
            "name": "Text Right of Icon",
            "type": "TextField",
            "value": "",
          },
          Object {
            "name": "Subtitle",
            "type": "TextBox",
            "value": "",
          },
        ],
      },
    ],
  },
]
`;

exports[`App when text only and when starting with a preview prepares the data for order as expected 1`] = `
Array [
  Object {
    "audioInfo": Object {
      "audioType": "StockAudio",
      "audioUrl": "https://fpsound.s3.amazonaws.com/13.mp3",
      "id": 13,
      "length": 3,
      "name": "Bunny Garden",
    },
    "imageBank": Array [],
    "isPreview": true,
    "resolutionId": 5,
    "resolutionOptions": Array [
      Object {
        "id": 5,
        "name": "720p",
      },
      Object {
        "id": 3,
        "name": "1080p",
      },
    ],
    "ui": Array [
      Object {
        "Customize Icon": Array [
          Object {
            "name": "Icon Style",
            "type": "DropDown",
            "value": "camera",
          },
          Object {
            "name": "Icon Character - type in only one (optional)",
            "type": "TextField",
            "value": "q",
          },
        ],
      },
      Object {
        "Customize Look": Array [
          Object {
            "name": "YT1",
            "type": "YouTubeLink",
            "value": "The Office Audition Tapes For Dwight, Michael, Kevin, Pam and Jim|S73Nzksy6rU|",
          },
          Object {
            "name": "Icon Color",
            "type": "DropDown",
            "value": "icon_pink",
          },
          Object {
            "name": "Main Text Color",
            "type": "ColorPicker",
            "value": "100.100.0",
          },
          Object {
            "name": "Subtitle Text Color",
            "type": "ColorPicker",
            "value": "0.0.0.50",
          },
        ],
      },
      Object {
        "Customize Text": Array [
          Object {
            "name": "Text Left of Icon",
            "type": "TextField",
            "value": "Left",
          },
          Object {
            "name": "Text Right of Icon",
            "type": "TextField",
            "value": "Right",
          },
          Object {
            "name": "Subtitle",
            "type": "TextBox",
            "value": "Bottom",
          },
        ],
      },
    ],
  },
]
`;
